### PR TITLE
Plat UI 2209 govuk frontend 4 5 0

### PR DESCRIPTION
**Pull request by JoPintoPaul**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,21 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
+### Changed
+
+- Uplifted version of `govuk-frontend` to `4.5.0`
+- Added new summary `Card` functionality to `SummaryList`, as implemented in `govuk-frontend`
+- Patched tests for `GovukTemplate`. `govuk-frontend` now allows passing in of custom Opengraph URL, but at this time we
+have decided not to replicate in `play-frontend-hmrc`
+
+### Compatible with
+
+- [hmrc/hmrc-frontend v5.23.0](https://github.com/hmrc/hmrc-frontend/releases/tag/v5.23.0)
```
Comment (ellamdav) [inline CHANGELOG.md:22]: should be `v5.24.0` ?

File: src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html [hunk 2]
```diff
@@ -14,11 +14,46 @@
...
-<dl class="@toClasses("govuk-summary-list", classes)"@toAttributes(attributes)>
+
+@summaryCard(card: Card) = {
+  <div class="@toClasses("govuk-summary-card", card.classes)"@toAttributes(card.attributes)>
+    <div class="govuk-summary-card__title-wrapper">
+      @if(card.title.isDefined) {
+      <h@{card.title.get.headingLevel.getOrElse(2)} class="@toClasses("govuk-summary-card__title", card.title.get.classes)">
+        @card.title.get.content.asHtml
+      </h@{card.title.get.headingLevel.getOrElse(2)}>
+       }
```
Comment (ellamdav) [inline src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html:31]: would this work?
```suggestion
      @card.title.map { title =>
      <h@{title.headingLevel.getOrElse(2)} class="@toClasses("govuk-summary-card__title", title.classes)">
        @title.content.asHtml
      </h@{title.headingLevel.getOrElse(2)}>
       }
```

File: src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html [hunk 3]
```diff
@@ -14,11 +14,46 @@
...
+       } else {
+         <ul class="@toClasses("govuk-summary-card__actions", card.actions.get.classes)">
+         @card.actions.map(_.items).getOrElse(Seq.empty).map { item =>
+           <li class="govuk-summary-card__action">
+             @actionLink(item)
+           </li>
+         }
+         </ul>
+       }
+     }
```
Comment (ellamdav) [inline src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html:46]: similarly something like this (untested)
```suggestion
     @card.actions match {
       case None =>
       case Some(actions) => actions.items match {
         case Nil =>
         case item :: Nil =>
           <div class="@toClasses("govuk-summary-card__actions", actions.classes)">
             @actionLink(item)
           </div>
         case items =>
           <ul class="@toClasses("govuk-summary-card__actions", actions.classes)">
             @items.map { item =>
               <li class="govuk-summary-card__action">
                 @actionLink(item)
               </li>
             }
           </ul>
       }
     }
```

File: src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html (hunk 2 above)
Comment (JoPintoPaul) [reply src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html]: @ellamdav Perfect, good suggestion, rewritten as per your proposal.

File: src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html (hunk 3 above)
Comment (JoPintoPaul) [reply src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukSummaryList.scala.html]: @ellamdav I had issues with nested case matching in Twirl - one level of case statements seems fine but the second level was having issues compiling - however I have used one level of case matching to make the code significantly cleaner, be great to get your feedback.

File: CHANGELOG.md (hunk 1 above)
Comment (JoPintoPaul) [reply CHANGELOG.md]: @ellamdav Corrected, with thanks.

**Review by ellamdav — APPROVED** [review]

**Pull request merged**